components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/ConnectorResource.java

```diff
@@ -0,0 +1,203 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.micro.integrator.management.apis;
+
+import com.google.gson.JsonObject;
+import org.apache.axis2.AxisFault;
+import org.apache.axis2.engine.AxisConfiguration;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.commons.json.JsonUtil;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.libraries.imports.SynapseImport;
+import org.apache.synapse.libraries.model.Library;
+import org.apache.synapse.libraries.model.SynapseLibrary;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.mediation.initializer.ServiceBusUtils;
+import org.wso2.carbon.mediation.initializer.persistence.MediationPersistenceManager;
+import org.wso2.carbon.micro.integrator.core.deployment.synapse.deployer.SynapseAppDeployer;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.wso2.carbon.micro.integrator.management.apis.Constants.ITEM_TYPE_IMPORT;
+
+/**
+ * API Resource to manage connectors deployed
+ **/
+public class ConnectorResource implements MiApiResource {
+
+    private static final Log LOG = LogFactory.getLog(ConnectorResource.class);
+
+    // Set of HTTP methods allowed by the resource
+    Set<String> methods = null;
+    // Attributes in json payload
+    private static final String NAME_ATTRIBUTE = "name";
+    private static final String STATUS_ATTRIBUTE = "status";
+    private static final String PACKAGE_ATTRIBUTE = "package";
+    private static final String DESCRIPTION_ATTRIBUTE = "description";
+
+    public ConnectorResource() {
+
+        methods = new HashSet<>();
+        methods.add(Constants.HTTP_POST);
+        methods.add(Constants.HTTP_GET);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext,
+                          org.apache.axis2.context.MessageContext axis2MessageContext,
+                          SynapseConfiguration synapseConfiguration) {
+
+        if (messageContext.isDoingGET()) {
+            populateConnectorList(axis2MessageContext, synapseConfiguration);
+            axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        } else if (Utils.isDoingPOST(axis2MessageContext)) {
```

Comment:
we do not need to check POST since only options are GET and POST

Reply:
+1 Resolved